workitem.go

```diff
@@ -160,43 +120,27 @@ func (c *WorkitemController) Create(ctx *app.CreateWorkitemContext) error {
 		jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrUnauthorized(err.Error()))
 		return ctx.Unauthorized(jerrors)
 	}
-
 	var wit *string
-	if ctx.Payload.Data != nil && ctx.Payload.Data.Relationships != nil && ctx.Payload.Data.Relationships.BaseType != nil {
-		if ctx.Payload.Data.Relationships.BaseType.Data != nil {
-			wit = &ctx.Payload.Data.Relationships.BaseType.Data.ID
-		}
+	if ctx.Payload.Data != nil && ctx.Payload.Data.Relationships != nil &&
+		ctx.Payload.Data.Relationships.BaseType != nil && ctx.Payload.Data.Relationships.BaseType.Data != nil {
+		wit = &ctx.Payload.Data.Relationships.BaseType.Data.ID
 	}
 	if wit == nil { // TODO Figure out path source etc. Should be a required relation
-		jerrors, _ := jsonapi.ErrorToJSONAPIErrors(errors.NewBadParameterError("data.relationships.basetype.data.id", nil))
-		return ctx.BadRequest(jerrors)
-
+		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("Missing Data.Relationships.BaseType.Data.ID element in request", err))
```

Comment:
I don't like the way you "misuse" the `param` attribute of the `NewBadParameterError` function. This is supposed to be the parameter and not an error explanation. This would be my approach once the API is refactored as I described above:

```go
return jsonapi.JSONErrorResponse(ctx,
    errors.NewBadParameterError(errs.New("Failed to get WIT"), "Data.RelationShips.BaseType.Data.ID", nil),
)
```